roles/openshift_grafana/defaults/main.yaml

```diff
@@ -0,0 +1,54 @@
+---
+
+openshift_grafana_state: present
+openshift_grafana_namespace: grafana
+openshift_grafana_pod_timeout: 300
+openshift_grafana_prometheus_namespace: "{{ openshift_prometheus_namespace | default('openshift-metrics') }}"
+openshift_grafana_prometheus_serviceaccount: prometheus
+openshift_grafana_serviceaccount_name: grafana
+openshift_grafana_serviceaccount_annotations: []
+l_openshift_grafana_serviceaccount_annotations:
+  - serviceaccounts.openshift.io/oauth-redirectreference.primary='{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"grafana"}}'
+openshift_grafana_datasource_name: "prometheus"
+openshift_grafana_node_exporter: false
+openshift_grafana_graph_granularity: "2m"
+openshift_grafana_node_selector: {"region":"infra"}
+openshift_grafana_hostname: grafana-{{openshift_grafana_namespace}}.{{openshift_master_default_subdomain}}
```

Comment:
Same here, from some reason the route I've got was no-responsive for me, so I use it empty as Introduced in the origin example https://github.com/openshift/origin/blob/master/examples/grafana/grafana.yaml#L19-L21
and it just works, and I've got responsive route
grafana-grafana.apps.0220-p0c.qe.<...>.com

Reply:
does this fail? as we use same in Prometheus and glusterfs plays

Reply:
I've run this grafana playbook in particularly.

Reply:
BTW, my prometheus has the similar subdomain:
prometheus-openshift-metrics.apps.0220-p0c.qe.<...>.com

@mffiedler @mbruzek 
any insights ?

Reply:
Openahift-orgin playbook prometheus?

Reply:
As this follows this:
https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_prometheus/defaults/main.yaml#L8-L10

Reply:
Yep, I think our jenkins runs it a bit differently, that why I've cced Mike and Matt.

Also, I just tried to run prometheus manually, and thats what I've got (also for grafana BTW)
fatal: [localhost]: FAILED! => {"changed": false, "msg": "No schedulable nodes found matching node selector for Prometheus - '{u'region': u'primary'}'"}

I changed it "primary" but it happens when using infra too.
this is something new (the "Ensure that Prometheus has nodes to run on" task)?!

Reply:
is it 3.7 or 3.9? Do you have these nodes available?

Reply:
Also, when deploy it manually I see the route now with the subdomain.

Reply:
Ansible module which enabled "node availability" validation was introduced only in 3.9 code. So there might be some misalignment if you use older version of modules. But overcall from the error looks like node label is missing?

Reply:
I think is more about our aws inventory, I just realized we use the following:
openshift_master_default_subdomain_enable=true
openshift_master_default_subdomain=apps.<*>.<*>.com

so I just use it that way and it works.

Reply:
What about node availability part?